STEER/STEERBase/AliTPCPIDResponse.cxx

```diff
@@ -2447,3 +2447,109 @@ double AliTPCPIDResponse::sigmadEdxPt(double p, double PID, double resol ){
   double deltaRel = TMath::Abs(dEdx2-dEdx)/dEdx;
   return deltaRel;
 }
+
+
+/// return pt after traversing length in TPC volume assummig AliExternalTrackParam::BetheBlochAleph standard parameterization
+/// formula approximation for primary and new to primary tracks
+/// \param ptIn       - input Pt
+/// \param tgl        - pz/pt
+/// \param mass       - particle mass
+/// \param type       - 0-Argon, 1-Neon
+/// \param bz         - magnetic field in kG
+/// \param fraction   - scaling factor for energy loss
+/// \return
+double AliTPCPIDResponse::GetPtOutHelix(Float_t ptIn, Double_t tgl, Float_t mass, Double_t rIn, Double_t rOut,  Int_t type, Float_t bz,Float_t fraction){
+  const Float_t kB2C=-0.299792458e-3;
+  const Float_t kMaxSnp=0.95;
+  Float_t pin=ptIn*TMath::Sqrt(1+tgl*tgl);
+  Float_t qPt=1/ptIn;
+  Float_t bg = pin/mass;
+  Float_t elossNe = 0.001*0.00156;  // page 3 https://www.slac.stanford.edu/pubs/icfa/summer98/paper3/paper3.pdf
+  Float_t elossAr = 0.001*0.00244;  // page 3 https://www.slac.stanford.edu/pubs/icfa/summer98/paper3/paper3.pdf
+  Float_t dEdx= AliExternalTrackParam::BetheBlochAleph(bg)*fraction;
+  Double_t e0=TMath::Sqrt(pin*pin+mass*mass);
+  Float_t eloss=(type==0) ? elossAr:elossNe;
+  Float_t lengthNorm=TMath::Abs(rOut-rIn)*TMath::Sqrt(1+tgl*tgl);
+  Float_t C = kB2C*bz*qPt;
+  if (TMath::Abs(0.5*rIn*C)>1)  rIn=2.*kMaxSnp/C;
+  if (TMath::Abs(0.5*rOut*C)>1)  rOut=2.*kMaxSnp/C;
+  Float_t lIn = 2.*TMath::ASin(0.5*rIn*C)/C;
+  Float_t lOut = 2.*TMath::ASin((0.5*rOut*C))/C;
+  lengthNorm=(TMath::Abs(lOut)-TMath::Abs(lIn))*TMath::Sqrt(1+tgl*tgl);
+   //
+  Double_t e1=e0-dEdx*eloss*lengthNorm;
+  if (e1<mass) return 0;
+  return TMath::Sqrt(e1*e1-mass*mass)/TMath::Sqrt(1+tgl*tgl);
+}
```

Comment:
- Why mix float and double precision?
- Is double needed? If not I propose to make all float otherwise all double.
- nearly all values can be const.
- Why is lengthNorm overwritten? The first value is not used.